HW1/P2/P2b.py

```diff
@@ -1,3 +1,34 @@
 from P2 import *
 
-# Your code here
+# Initialize Spark
+from pyspark import SparkContext
+sc = SparkContext()
+sc.setLogLevel("ERROR")
+
+# Define key parameters
+coordWidth = 2000
+coordHeight = 2000
+noPartitions = 100
+
+# Create RDD with i, j coordinates
+pixCoordIJ = sc.parallelize([(x, y) for x in range(coordWidth) for y in range(coordHeight)])
+
+# Apply the Mandelbrot function & set number of partitions
+pixMandel = pixCoordIJ.map(lambda (i, j): ((i, j), mandelbrot(j/500.0 - 2, i/500.0 - 2))).partitionBy(noPartitions)
```

Comment:
Let's see if I can clarify the confusion!  The choice of partitionBy was specific to this problem, where you can see from the imagine that the jobs with the highest number of iterations are clustered in the center of the plot. My goal was not to come up with a catch-all partition function, but to find an approach that would work given these specific key-value pairs.

@btweinstein : Yes, my understanding is that partitionBy will do a simple hash partition if it is not passed a specific function. However, in this case the hash partition has done a good job of breaking up that cluster of high-iteration jobs, which is why I went with this approach. 

@Sdeppen : Yes, the graphs were pretty consistent when re-running the problem several times.